compiler/rustc_ast_lowering/src/item.rs

```diff
@@ -1657,6 +1671,32 @@ impl<'hir> LoweringContext<'_, 'hir> {
         (lowered_generics, res)
     }
 
+    pub(super) fn lower_define_opaques(
+        &mut self,
+        define_opaques: &Option<ThinVec<(NodeId, Path)>>,
+    ) -> Option<hir::Attribute> {
```

Comment:
so what we do is:
- ast has a builtin attr
- this attribute adds the opaque path to a field of the fn in the ast
- name resolution then resolves this field to the `DefId` of the opaque
- ast -> hir then moves these `DefId` back as an attribute of the `fn`
 
Why do we represent this via attributes on the function, wouldn't it be easier to feed a `query defined_opaque_types` or sth like that for the function?

Reply:
Uh 🤯 I'll think about that. It has some issues, but they are probably resolvable

Reply:
an alternative would be to keep it as a field of the mir. It just feels somewhat weird to me to go back to HIR attributes after already converting it to something builtin on the AST

Reply:
We can't feed the `defined_opaque_types` query as it also needs to walk the signature for RPITs. So we'd need a second query to feed. I instead just put the information into the general HIR that exists per item